cleverhans/train.py

```diff
@@ -104,78 +106,122 @@ def train(sess, loss, x_train, y_train,
 
             if x_batch_preprocessor is not None:
                 x = x_batch_preprocessor(x)
+
+            # We need to keep track of these so that the canary can feed
+            # preprocessed values. If the canary had to feed raw values,
+            # stochastic preprocessing could make the canary fail.
+            preprocessed_xs.append(x)
+
+            loss.model.get_params()
             loss_value = loss.fprop(x, y, **fprop_args)
+            loss.model.get_params()
```

Comment:
Could you clarify why the second call to get_params() is needed here? Is it impacted by fprop?

Reply:
That was debugging code that I didn't mean to commit. I've removed it. Actually neither call to get_params is needed. I just had them there for debugging purposes, to make sure the model I was working with didn't add new parameters when I ran it on each new GPU.

Reply:
I understand, thanks for the explanation